get_element_cpo/P2769.bs

```diff
@@ -429,150 +438,128 @@ wider range of types rather than just the listed standard types.
-i.e., `std::ranges::get<Iarg, Sarg, Karg>(std::ranges::subrange<Iarg, Sarg, Karg>{})`. The problem is
-with the last `Karg` argument, which is unrelated to `tuple_size_v` and `tuple_element_t` of the subrange.
-Even if we say that the proposed backward compatible CPO with `Args...` does not constraint `sizeof...(Args)`
-and ignores the tail outside `tuple_size_v<subrange>`, it doesn't help
-for the mentioned use case because `K` of `std::ranges::subrange` is a non-type template parameter.
-Anyway, this scenario doesn't look common because explicit template parameters are used relatively rarely
-and furthermore, `K` has the default argument that is substituted based on a sentinel.
-
-Definitely such a change would break the ABI for `std::ranges::get` because the fully qualified name of
-this function would change from what is in C++ standard library implementations today.
-But we think that that ABI break would have low impact because `std::ranges::get` is likely
-inlined and so doesn't create a visible symbol in translation units.
-We could imagine other tricky scenario where the API might be broken when `std::ranges::get` is used for something else but call. For example:
-`&std::ranges::get<0>`, `decltype(std::ranges::get<0>)`, etc. However, these scenarios don't look common.
-
-Since the `std::ranges::subrange` API is relatively new, perhaps only a small amount
-of users would be affected but it can not be predicted accurately.
+However, such potential change is very unlikely. In particular, there is no user-defined type involved. If such code was deployed, it's likely to be already broken. Indeed, while `std::get` is not a customisation point, it is often used unqualified (and therefore found by ADL), such that the code above can already be ambiguous.
+Therefore we do not think that this potential breaking change requires any preventive measure (and we don't think such measure exist)
+about the scenario mentioned above.
+
+## `get_element` for `std::variant` (or other non-tuple-like) ## {#get_element_std_variant}
+
+During the discussion in LEWG (St. Louis, 2024) we got a feedback that it's better to constraint `get_element` with
+`tuple_size` and possibly `tuple_element`. The idea was to not use it for something that is not `tuple-like`, e.g.,
+`std::variant`.
+
+While it might be good for `get_element` to be applicable in the same context where `std::get` is used it's not like that
+with the current proposal (purposefully):
+- `std::get` has overloads that allow to access `tuple` elements by type, while `std::get_element` doesn't.
+- It hard to imagine the generic code that expects both `tuple-like` or `std::variant` as an argument because it's always
+    safe to access any element of a `tuple-like` object whereas for `std::variant` only one alternative is "active" - and therefore safe to access. Supporting variant in this interface would make it easy to introduce UB in generic code.
```

Comment:
The same argument applies to `std::get` though, right? Do we therefore say that allowing `std::get` to be used with `std::variant` was a mistake and we do not want to repeat it?

Reply:
Reworked to say about preventing possible UBs.